### PR TITLE
issue #799 2.10 cross build

### PR DESCRIPTION
**Pull request by jho**
Impressive work guys.  I am stuck working with 2.10 because of some external dependencies.  Was hoping that there is interest in maintaining backwards compatibility with 2.10 for the near term.

[Review on Reviewable](https://reviewable.io/reviews/intel-hadoop/gearpump/794)

### CODE REVIEW

**Comment by clockfly** [comment]
@jho 

Thanks for your contribution, would you mind openning a new issue here: https://github.com/intel-hadoop/gearpump/issues
and add the issue id in PR title, and commit log.

**Comment by huafengw** [comment]
Hi, Joshua
I tried your PR but seems not every project is compiled with scala-2.10, can you verify this? Thanks.

**Comment by jho** [comment]
If I compile in sbt with "+compile" I see it compiling all the sub projects.  They also publish locally for me with "+publish-local".

I'm new to trying to cross build something so I'm not sure if there is another way, but thats what has worked for me.

Also locally the tests pass as well.  Not sure why travis keeps failing.

File: examples/streaming/complexdag/src/main/scala/org/apache/gearpump/streaming/examples/complexdag/Node.scala [hunk 1]
```diff
@@ -22,13 +22,16 @@ import org.apache.gearpump.Message
...
+import scala.collection.mutable.ListBuffer
+
 class Node (taskContext : TaskContext, conf: UserConfig) extends Task(taskContext, conf) {
   import taskContext.output
 
   override def onStart(startTime : StartTime) : Unit = {}
 
   override def onNext(msg : Message) : Unit = {
-    val list = msg.msg.asInstanceOf[List[String]]
+    val list = msg.msg.asInstanceOf[Vector[String]]
```
Comment (clockfly) [inline examples/streaming/complexdag/src/main/scala/org/apache/gearpump/streaming/examples/complexdag/Node.scala]: Why you need to change the type?

**Comment by clockfly** [comment]
@jho 

The build failure seems random. By the way, why you change the ```List`` to ```Vector```?

File: examples/streaming/complexdag/src/main/scala/org/apache/gearpump/streaming/examples/complexdag/Node.scala (hunk 1 above)
Comment (jho) [inline examples/streaming/complexdag/src/main/scala/org/apache/gearpump/streaming/examples/complexdag/Node.scala]: List is not Serializable in 2.10.

**Comment by huafengw** [comment]
Hi, @jho 
Fow now Gearpump's Build will only generate one .tar.gz, which is scala-2.11 version.
so if this pull request can solve your problem you met with scala-2.10, I'll merge it.

**Comment by clockfly** [comment]
+1

**Comment by jho** [comment]
If you guys can accept this pull request for just building 2.10, I'll start working on figuring out how to produce a package for 2.10 for another issue.  I'll need both long term to make it all work, but just having the libraries build with 2.10 is a good start.  Thanks.

**Comment by huafengw** [comment]
Hi @jho 
After changing List to Vector, you should add a config in core/src/main/resource.conf, for section gearpump.serializers, add ```"scala.collection.immutable.Vector" = ""```
Please update your PR and I'll merge it then. Thanks.

**Comment by clockfly** [comment]
@huafengw 

Let's make the change instead.

**Comment by clockfly** [comment]
Thanks to Jho for the contribution.

**Pull request merged**